### ISSUE TITLE
Unable to start an Azure ML compute using az ml cli v2

### ISSUE THREAD

**Issue by vladiliescu** [Azure/azure-cli-extensions#5588]
## Unable to start an Azure ML compute using az ml cli v2

**Command Name**
`az ml compute start
Extension Name: ml. Version: 2.11.0.`

**Errors:**
```
Met error <class 'TypeError'>:ComputeOperations.begin_start() got an unexpected keyword argument 'no_wait'
Please check log by running the command with '--debug' for more details.

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- Create an az ml compute
- Try to start it using `az ml compute start --name {} -g {} -w {} --no-wait --debug`

## Expected Behavior

- The compute starts, with no cli errors whatsoever

## Environment Summary
```
macOS-12.6.1-x86_64-i386-64bit, Darwin 21.6.0
Python 3.10.8
Installer: HOMEBREW

azure-cli 2.42.0

Extensions:
ml 2.11.0
account 0.2.5

Dependencies:
msal 1.20.0
azure-mgmt-resource 21.1.0b1
```

**Comment by yonzhan** [comment]
route to CXP team

**Comment by navba-MSFT** [comment]
@vladiliescu Could you please run the same command again with ` --debug` parameter and send the error output ? Awaiting your reply.

**Comment by vladiliescu** [comment]
Sure @navba-MSFT, here you go

```
cli.knack.cli: Command arguments: ['ml', 'compute', 'start', '-n', '--redacted--', '-w', '--redacted--', '-g', '--redacted--', '--debug']
cli.knack.cli: __init__ debug log:
Enable color in terminal.
cli.knack.cli: Event: Cli.PreExecute []
cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x109bb0550>, <function OutputProducer.on_global_arguments at 0x109c15ab0>, <function CLIQuery.on_global_arguments at 0x109c46ef0>]
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
cli.azure.cli.core: Modules found from index for 'ml': ['azext_mlv2']
cli.azure.cli.core: Loading command modules:
cli.azure.cli.core: Name                  Load Time    Groups  Commands
cli.azure.cli.core: Total (0)                 0.000         0         0
cli.azure.cli.core: These extensions are not installed and will be skipped: ['azext_ai_examples', 'azext_next']
cli.azure.cli.core: Loading extensions:
cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
azext_mlv2.manual.vendored_curated_sdk.azure.ai.ml._azure_environments: Using the default cloud configuration: 'AzureCloud'.
azext_mlv2.manual.vendored_curated_sdk.azure.ai.ml._azure_environments: Using the cloud configuration: '{'azure_portal': 'https://portal.azure.com/', 'resource_manager': 'https://management.azure.com/', 'active_directory': 'https://login.microsoftonline.com/', 'aml_resource_id': 'https://ml.azure.com/', 'storage_endpoint': 'core.windows.net'}'.
azext_mlv2.manual.vendored_curated_sdk.azure.ai.ml._azure_environments: Using the default cloud configuration: 'AzureCloud'.
azext_mlv2.manual.vendored_curated_sdk.azure.ai.ml._azure_environments: Using the cloud configuration: '{'azure_portal': 'https://portal.azure.com/', 'resource_manager': 'https://management.azure.com/', 'active_directory': 'https://login.microsoftonline.com/', 'aml_resource_id': 'https://ml.azure.com/', 'storage_endpoint': 'core.windows.net'}'.
azext_mlv2.manual.vendored_curated_sdk.azure.ai.ml._azure_environments: Using the default cloud configuration: 'AzureCloud'.
azext_mlv2.manual.vendored_curated_sdk.azure.ai.ml._azure_environments: Using the cloud configuration: '{'azure_portal': 'https://portal.azure.com/', 'resource_manager': 'https://management.azure.com/', 'active_directory': 'https://login.microsoftonline.com/', 'aml_resource_id': 'https://ml.azure.com/', 'storage_endpoint': 'core.windows.net'}'.
cli.azure.cli.core: ml                        2.169        16       105  /Users/vladiliescu/.azure/cliextensions/ml
cli.azure.cli.core: Total (1)                 2.169        16       105
cli.azure.cli.core: Loaded 16 groups, 105 commands.
cli.azure.cli.core: Found a match in the command table.
cli.azure.cli.core: Raw command  : ml compute start
cli.azure.cli.core: Command table: ml compute start
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x10a72d090>]
cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to '/Users/vladiliescu/.azure/commands/2022-12-02.11-10-26.ml_compute_start.4812.log'.
az_command_data_logger: command args: ml compute start -n {} -w {} -g {} --debug
cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x10a805900>]
cli.knack.commands: Configured default '--redacted--' for arg resource_group_name
cli.knack.commands: Configured default '--redacted--' for arg workspace_name
cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x10a8283a0>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x10a828430>]
cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
cli.knack.cli: Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x109c15b40>, <function CLIQuery.handle_query_parameter at 0x109c46f80>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x10a828310>]
az_command_data_logger: extension name: ml
az_command_data_logger: extension version: 2.11.0
cli.azure.cli.core.commands.client_factory: Getting management service client client_type=AzureMachineLearningWorkspaces
cli.azure.cli.core.auth.persistence: build_persistence: location='/Users/vladiliescu/.azure/msal_token_cache.json', encrypt=False
cli.azure.cli.core.auth.binary_cache: load: /Users/vladiliescu/.azure/msal_http_cache.bin
urllib3.util.retry: Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
msal.authority: openid_config = {'token_endpoint': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/oauth2/v2.0/token', 'token_endpoint_auth_methods_supported': ['client_secret_post', 'private_key_jwt', 'client_secret_basic'], 'jwks_uri': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/discovery/v2.0/keys', 'response_modes_supported': ['query', 'fragment', 'form_post'], 'subject_types_supported': ['pairwise'], 'id_token_signing_alg_values_supported': ['RS256'], 'response_types_supported': ['code', 'id_token', 'code id_token', 'id_token token'], 'scopes_supported': ['openid', 'profile', 'email', 'offline_access'], 'issuer': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/v2.0', 'request_uri_parameter_supported': False, 'userinfo_endpoint': 'https://graph.microsoft.com/oidc/userinfo', 'authorization_endpoint': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/oauth2/v2.0/authorize', 'device_authorization_endpoint': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/oauth2/v2.0/devicecode', 'http_logout_supported': True, 'frontchannel_logout_supported': True, 'end_session_endpoint': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/oauth2/v2.0/logout', 'claims_supported': ['sub', 'iss', 'cloud_instance_name', 'cloud_instance_host_name', 'cloud_graph_host_name', 'msgraph_host', 'aud', 'exp', 'iat', 'auth_time', 'acr', 'nonce', 'preferred_username', 'name', 'tid', 'ver', 'at_hash', 'c_hash', 'email'], 'kerberos_endpoint': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/kerberos', 'tenant_region_scope': 'EU', 'cloud_instance_name': 'microsoftonline.com', 'cloud_graph_host_name': 'graph.windows.net', 'msgraph_host': 'graph.microsoft.com', 'rbac_url': 'https://pas.windows.net'}
msal.application: Broker enabled? False
cli.azure.cli.core.commands.client_factory: Getting management service client client_type=_ml_client_cli
urllib3.util.retry: Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
msal.authority: openid_config = {'token_endpoint': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/oauth2/v2.0/token', 'token_endpoint_auth_methods_supported': ['client_secret_post', 'private_key_jwt', 'client_secret_basic'], 'jwks_uri': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/discovery/v2.0/keys', 'response_modes_supported': ['query', 'fragment', 'form_post'], 'subject_types_supported': ['pairwise'], 'id_token_signing_alg_values_supported': ['RS256'], 'response_types_supported': ['code', 'id_token', 'code id_token', 'id_token token'], 'scopes_supported': ['openid', 'profile', 'email', 'offline_access'], 'issuer': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/v2.0', 'request_uri_parameter_supported': False, 'userinfo_endpoint': 'https://graph.microsoft.com/oidc/userinfo', 'authorization_endpoint': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/oauth2/v2.0/authorize', 'device_authorization_endpoint': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/oauth2/v2.0/devicecode', 'http_logout_supported': True, 'frontchannel_logout_supported': True, 'end_session_endpoint': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/oauth2/v2.0/logout', 'claims_supported': ['sub', 'iss', 'cloud_instance_name', 'cloud_instance_host_name', 'cloud_graph_host_name', 'msgraph_host', 'aud', 'exp', 'iat', 'auth_time', 'acr', 'nonce', 'preferred_username', 'name', 'tid', 'ver', 'at_hash', 'c_hash', 'email'], 'kerberos_endpoint': 'https://login.microsoftonline.com/--guess-its-redacted-as-well--/kerberos', 'tenant_region_scope': 'EU', 'cloud_instance_name': 'microsoftonline.com', 'cloud_graph_host_name': 'graph.windows.net', 'msgraph_host': 'graph.microsoft.com', 'rbac_url': 'https://pas.windows.net'}
msal.application: Broker enabled? False
Traceback (most recent call last):
  File "/Users/vladiliescu/.azure/cliextensions/ml/azext_mlv2/manual/custom/compute.py", line 163, in ml_compute_start
    compute = ml_client.compute.begin_start(name=name, no_wait=no_wait)
  File "/usr/local/Cellar/azure-cli/2.42.0/libexec/lib/python3.10/site-packages/azure/core/tracing/decorator.py", line 73, in wrapper_use_tracer
    return func(*args, **kwargs)
  File "/Users/vladiliescu/.azure/cliextensions/ml/azext_mlv2/manual/vendored_curated_sdk/azure/ai/ml/_telemetry/activity.py", line 259, in wrapper
    return f(*args, **kwargs)
TypeError: ComputeOperations.begin_start() got an unexpected keyword argument 'no_wait'
cli: None
cli: Traceback (most recent call last):
  File "/Users/vladiliescu/.azure/cliextensions/ml/azext_mlv2/manual/custom/compute.py", line 163, in ml_compute_start
    compute = ml_client.compute.begin_start(name=name, no_wait=no_wait)
  File "/usr/local/Cellar/azure-cli/2.42.0/libexec/lib/python3.10/site-packages/azure/core/tracing/decorator.py", line 73, in wrapper_use_tracer
    return func(*args, **kwargs)
  File "/Users/vladiliescu/.azure/cliextensions/ml/azext_mlv2/manual/vendored_curated_sdk/azure/ai/ml/_telemetry/activity.py", line 259, in wrapper
    return f(*args, **kwargs)
TypeError: ComputeOperations.begin_start() got an unexpected keyword argument 'no_wait'

cli.azure.cli.core.util: azure.cli.core.util.handle_exception is called with an exception:
cli.azure.cli.core.util: Traceback (most recent call last):
  File "/Users/vladiliescu/.azure/cliextensions/ml/azext_mlv2/manual/custom/compute.py", line 163, in ml_compute_start
    compute = ml_client.compute.begin_start(name=name, no_wait=no_wait)
  File "/usr/local/Cellar/azure-cli/2.42.0/libexec/lib/python3.10/site-packages/azure/core/tracing/decorator.py", line 73, in wrapper_use_tracer
    return func(*args, **kwargs)
  File "/Users/vladiliescu/.azure/cliextensions/ml/azext_mlv2/manual/vendored_curated_sdk/azure/ai/ml/_telemetry/activity.py", line 259, in wrapper
    return f(*args, **kwargs)
TypeError: ComputeOperations.begin_start() got an unexpected keyword argument 'no_wait'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.42.0/libexec/lib/python3.10/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/Cellar/azure-cli/2.42.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/usr/local/Cellar/azure-cli/2.42.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/local/Cellar/azure-cli/2.42.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
  File "/usr/local/Cellar/azure-cli/2.42.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/local/Cellar/azure-cli/2.42.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/Users/vladiliescu/.azure/cliextensions/ml/azext_mlv2/manual/custom/compute.py", line 168, in ml_compute_start
    log_and_raise_error(err, debug)
  File "/Users/vladiliescu/.azure/cliextensions/ml/azext_mlv2/manual/custom/raise_error.py", line 145, in log_and_raise_error
    raise cli_error
knack.util.CLIError: Met error <class 'TypeError'>:ComputeOperations.begin_start() got an unexpected keyword argument 'no_wait'
Please check log by running the command with '--debug' for more details.

cli.azure.cli.core.azclierror: Met error <class 'TypeError'>:ComputeOperations.begin_start() got an unexpected keyword argument 'no_wait'
Please check log by running the command with '--debug' for more details.
az_command_data_logger: Met error <class 'TypeError'>:ComputeOperations.begin_start() got an unexpected keyword argument 'no_wait'
Please check log by running the command with '--debug' for more details.
cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x10a72d2d0>]
az_command_data_logger: exit code: 1
cli.__main__: Command ran in 2.452 seconds (init: 0.216, invoke: 2.237)
telemetry.main: Begin splitting cli events and extra events, total events: 1
telemetry.client: Accumulated 0 events. Flush the clients.
telemetry.main: Finish splitting cli events and extra events, cli events: 1
telemetry.save: Save telemetry record of length 3848 in cache
telemetry.check: Negative: The /Users/vladiliescu/.azure/telemetry.txt was modified at 2022-12-02 11:10:19.837194, which in less than 600.000000 s
```

**Comment by navba-MSFT** [comment]
@vladiliescu Thanks for the logs. Could you please run the same command with `--no-wait` parameter and check if that helps ?

Awaiting your reply.

**Comment by navba-MSFT** [comment]
@vladiliescu Apologies for the late reply. I am able to reproduce this issue at my end. I have reached out to the Product Group team for the fix. We have created an internal workitem for the fix. I will share the tentative ETA for the fix soon.

**Comment by navba-MSFT** [comment]
@vladiliescu The fix should be released in a couple of days in the new` ml `extension `2.12.0`. You can then update your extension using the below command and then test the command again. 

>az extension update -n ml

**Comment by navba-MSFT** [comment]
@vladiliescu Apologies for the late reply. Thanks for getting back. I am discussing this with the Product Owners again. I can reproduce this issue at my end too with the most recent ml extension. I will keep you posted once I hear from them.

**Comment by navba-MSFT** [comment]
@vladiliescu Apologies for the late reply. I have discussed this with our Product Owners and they are working on this. I will share the tentative eta for the fix once I hear from them. We appreciate your patience.

**Comment by navba-MSFT** [comment]
@vladiliescu The fix is released as a part of `2.12.1 `version of `ml `extension. Please update your `ml` extension by running the below command:

> **az extension update -n ml**

Once the extension is updated, Please stop your compute instance by running the command [az ml compute stop](https://learn.microsoft.com/en-us/cli/azure/ml/compute?view=azure-cli-latest#az-ml-compute-stop) and then you can run the [az ml compute start](https://learn.microsoft.com/en-us/cli/azure/ml/compute?view=azure-cli-latest#az-ml-compute-start)  command. 

**Note:** Start Azure ML Compute Instance is only allowed when the Compute instance is in 'stopped' state". 

I tried updating my extension and it worked fine at my end. Hope this helps.

**Comment by vladiliescu** [comment]
Thank you @navba-MSFT , I can confirm that it works on my side as well.
